src/browser/peer.ts

```diff
@@ -0,0 +1,72 @@
+import SimplePeer from "simple-peer";
+import { ISdpHandler } from "../base/sdp-handler";
+import { IStream } from "../base/stream-provider";
+import { IWebrtcProvider } from "../base/webrtc-provider";
+
+/**
+ * Peer constructor options
+ */
+export interface IPeerOpts {
+  /**
+   * The sdp handler
+   */
+  sdpHandler: ISdpHandler;
+}
+
+/**
+ * A peer webrtc provider
+ */
+export class Peer extends IWebrtcProvider {
+  private instance?: SimplePeer.Instance;
+  private sdpHandler: ISdpHandler;
+
+  /**
+   * Default ctor
+   * @param opts ctor opts
+   */
+  public constructor(opts: IPeerOpts) {
+    super();
+
+    this.sdpHandler = opts.sdpHandler;
+  }
+
+  public initialize(iceServers: RTCIceServer[], stream: IStream) {
+    if (this.instance) {
+      throw new Error("Already initialized");
+    }
+
+    // as SimplePeer needs <any> transformers, we must do that mapping
+    // for it
+    const anyTransformer = (sdp: any) => {
```

Comment:
this was needed as a "silly" typescript thing - SimplePeer expects `any: any` processor, we have `string: string` processor.